src/uct/base/uct_md.h

```diff
@@ -125,7 +128,7 @@ typedef ucs_status_t (*uct_md_detect_memory_type_func_t)(uct_md_h md,
  */
 struct uct_md_ops {
     uct_md_close_func_t                  close;
-    uct_md_query_func_t                  query;
+    uct_md_query_func_v2_t               query;
```

Comment:
do we need also a function to close the dmabuf handle, or is the assumption that it would be included in another operation (e.g. dereg?)

Reply:
`md_query_func` would only mention if the md can provide a dmabuf fd for the map of memory types in `.dmabuf_mem_types` of that md.

Unfortunately, according this PR, dmabuf fd is generated as part of `uct_md_mem_query` API which does not have resource release semantics. Ideally query API shouldn't have release operations but querying for dmabuf fd does have this side effect which appears ugly. Starting to think that it may be cleaner to have an API like `uct_md_mem_get_resource_handle`/`uct_md_mem_return_resource_handle` which could include dmabuf fd as a resource handle among other things. @yosefe @shamisp any thoughts on this?

Reply:
IMO since the user explicitly asks for dmabuf_fd by setting a field_mask bit, it's ok the user will close this fd using `close()` function. The alternative would be to have a separate function for every such resource we want to "get", which is uglier IMO.
BTW, `ucp_worker_query()` does the same thing with worker address.
Let's document the returned fd has to be `close()`-ed.

Reply:
Part of the problem is that the handle (at least in the rocm case) is not expected to be closed by a simple ```close()```, but by an internal function (in our case ```hsa_amd_portable_close_dmabuf()```). So we would have to maintain somewhere a pointer to the function that is expected to be invoked.

Reply:
Ouch. If there is a need for a special `close()` based on the TL implementation I don't see how we can easy workaround without creation of allocate/release wrapper.

Reply:
@edgargabriel Discussed offline with Yossi -- It seems like [ibv_reg_dmabuf_mr](https://github.com/linux-rdma/rdma-core/blob/master/libibverbs/verbs.h#L2571) requires a standard fd to be passed to it. This being the case, having these file descriptors be closed using standard API like `close()` seems like a requirement. Is it possible that rocm implementation can still conform to that? If yes, I think the suggestion that Yossi had [here](https://github.com/openucx/ucx/pull/7847#discussion_r882387805) will suffice.

Reply:
I will ask the runtime/kernel team

Reply:
I can confirm that using close() instead hsa_amd_portable_close_dmabuf() should work according to the kernel and runtime team on linux.